### PR TITLE
added contribution guidelines

### PR DESCRIPTION
**Pull request by bentheiii**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Before submitting your PR. You need to make sure it would pass some basic automation. For now that includes running the unit tests.
+```shell script
+#!/bin/bash
+# install poetry
+python -m pip install poetry # or your preferred method of installing poetry
+python -m poetry update --lock
+python -m poetry install -E redis -E rabbit -E kafka -E azure  # also include whatever new extras you added
+python -m poetry run python -m pytest --cov-branch tests/
+```
+The tests should run and pass for python 3.7 and 3.8, across Windows, IOS and Linux. You should also add new tests for whatever issue or feature you fixed/added, and add it to `CHANGELOG.md`.
```
Comment (aviramha) [inline CONTRIBUTING.md:16]: macOS

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Want to contribute? Great! First, read this page.
+
+# If there is no issue, open one
+Check if what you want to add/fix is documented in an issue. If not open one. Then, you can clone the repository and start hacking!
```
Comment (aviramha) [inline CONTRIBUTING.md:4]: Should be an option to skip an issue of the change is trivial.

**Review by aviramha — CHANGES_REQUESTED** [review]
Add a link to this from the README.

**Comment by bentheiii** [comment]
closes #21

File: CONTRIBUTING.md (hunk 2 above)
Comment (bentheiii) [reply CONTRIBUTING.md:4]: absolutely not, only admins should push to master

**Review by bharel — APPROVED** [review]

**Review by aviramha — APPROVED** [review]

**Pull request merged**